logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+ * @private
+ * @implements {__private.setAccountAndGet}
+ * @implements {__private.verifyTransaction}
+ * @param {transaction} transaction
+ * @param {function} cb - Callback function
+ * @returns {setImmediateCallback} errors | sender
+ */
+__private.processUnverifiedTransaction = function (transaction, cb) {
+	async.waterfall([
+		async.apply(__private.setAccountAndGet, transaction),
+		__private.verifyTransaction
+	], function (err, transaction, sender) {
+		return setImmediate(cb, err, sender);
+	});
+};
+
+// Public methods
+/**
+ * Bounds input parameters to private variable modules.
+ * @param {Accounts} accounts
+ */
+TransactionPool.prototype.bind = function (accounts) {
+	modules = {
+		accounts: accounts,
+	};
+};
+
+/**
+ * Gets invalid, unverified, verified.pending and verified.ready counters.
+ * @implements {__private.countTransactionsPool}
+ * @return {Object} unverified, pending, ready
+ */
+TransactionPool.prototype.getUsage = function () {
+	return {
+		unverified: pool.unverified.count,
+		pending: pool.verified.pending.count,
+		ready: pool.verified.ready.count,
+		invalid: pool.invalid.count,
+		total: __private.countTransactionsPool()
+	};
+};
+
+/**
+ * Gets transaction based on transaction id.
+ * Checks all pool lists: unverified, pending, ready.
+ * @param {string} id
+ * @return {Object} transaction, status
+ */
+TransactionPool.prototype.get = function (id) {
+	var transaction = pool.unverified.transactions[id];
+	if (transaction !== undefined) {
+		return {
+			transaction: transaction,
+			status: 'unverified'
+		};
+	}
+
+	transaction = pool.verified.pending.transactions[id];
+	if (transaction !== undefined) {
+		return {
+			transaction: transaction,
+			status: 'pending'
+		};
+	}
+	
+	transaction = pool.verified.ready.transactions[id];
+	if (transaction !== undefined) {
+		return {
+			transaction: transaction,
+			status: 'ready'
+		};
+	}
+	return {
+		transaction: undefined,
+		status: 'Transaction not in pool'
+	};
+};
+
+/**
+ * Gets all transactions based on limited filters.
+ * @implements {__private.getTransactionsFromPoolList}
+ * @implements {__private.getAllPoolTransactionsByFilter}
+ * @param {string} filter
+ * @param {Object} params
+ * @return {[transaction]} transaction list based on filter.
+ */
+TransactionPool.prototype.getAll  = function (filter, params) {
+	switch (filter) {
+		case 'unverified':
+			return __private.getTransactionsFromPoolList(pool.unverified.transactions, params.reverse, params.limit);
+		case 'pending':
+			return __private.getTransactionsFromPoolList(pool.verified.pending.transactions, params.reverse, params.limit);
+		case 'ready':
+			return __private.getTransactionsFromPoolList(pool.verified.ready.transactions, params.reverse, params.limit);
+		case 'sender_id':
+			return __private.getAllPoolTransactionsByFilter({'senderId': params.id});
+		case 'sender_pk':
+			return __private.getAllPoolTransactionsByFilter({'senderPublicKey': params.publicKey});
```

Comment:
We don't really need to repeat word `Pool` everywhere for private variables - because we're inside tx pool.